src/amberscript/parser.cc

```diff
@@ -1205,6 +1219,151 @@ Result Parser::ParseBuffer() {
   return {};
 }
 
+Result Parser::ParseImage() {
+  auto token = tokenizer_->NextToken();
+  if (!token->IsString())
+    return Result("invalid IMAGE name provided");
+
+  auto name = token->AsString();
+  if (name == "DATA_TYPE" || name == "FORMAT")
+    return Result("missing IMAGE name");
+
+  token = tokenizer_->NextToken();
+  if (!token->IsString())
+    return Result("invalid IMAGE command provided");
+
+  std::unique_ptr<Buffer> buffer;
```

Comment:
Could probably just do:

```
auto buffer = MakeUnique<Buffer>();
```

we'd waste the allocation in case of error, but otherwise we always make the buffer this way.